test/java/org/apache/ivy/core/resolve/ResolveTest.java

```diff
@@ -284,21 +287,56 @@ public void testResolveBadStatus() throws Exception {
         assertTrue(report.hasError());
     }
 
-    public void testResolveWithXmlEntities() throws Exception {
+    public void testResolveWithXmlEntities() {
+        testResolveWithXmlEntities(null, 0);
+        testResolveWithXmlEntities("prohibit", 0);
+        testResolveWithXmlEntities("ignore", 0);
+        testResolveWithXmlEntities("local-only", 2);
+        testResolveWithXmlEntities("LOCAL_ONLY", 2);
+        testResolveWithXmlEntities("all", 2);
+    }
+
+    private void testResolveWithXmlEntities(String externalResourcesSystemProperty,
+            int expectedNumberOfDependencies) {
         Ivy ivy = new Ivy();
         Throwable th = null;
+        Properties p = (Properties) System.getProperties().clone();
```

Comment:
@bodewig Creating a new empty `Properties` makes the test fail on Linux. So I cloned them instead.

Reply:
Hmm, I work on Linux. It may depend on the JVM version or vendor, though.